src/scripts/main.js

```diff
@@ -1,3 +1,261 @@
 'use strict';
 
-// write your code here
+const tbody = document.querySelector('tbody');
+const scoreField = document.querySelector('.game-score');
+
+const btnReset = document.createElement('button');
+
+const messageStart = document.querySelector('.message-start');
+const messageLose = document.querySelector('.message-lose');
+const messageWin = document.querySelector('.message-win');
+
+btnReset.classList.add('button');
+btnReset.classList.add('restart');
+btnReset.textContent = ('Restart');
+
+const rows = tbody.rows.length;
+const cells = rows;
+
+let board;
+let tempBoard;
+
+let score = 0;
+
+const matrix = [...tbody.children].map(row => [...row.children]);
+
+function getStart() {
+  btnStart.replaceWith(btnReset);
+
+  board = [
+    [0, 0, 0, 0],
+    [0, 0, 0, 0],
+    [0, 0, 0, 0],
+    [0, 0, 0, 0],
+  ];
+
+  score = 0;
+
+  messageStart.classList.add('hidden');
+  messageLose.classList.add('hidden');
+  messageWin.classList.add('hidden');
+
+  getRandomCell();
+  getRandomCell();
+
+  render();
+}
+
+function getRandomCell() {
+  const [y, x] = findEmptyCell();
+
+  board[y][x] = randomNumber();
+}
+
+function findEmptyCell() {
+  const emptyCells = [];
+
+  board.forEach((row, rowIndex) => {
+    row.forEach((cell, cellIndex) => {
+      if (cell === 0) {
+        emptyCells.push([rowIndex, cellIndex]);
+      }
+    });
+  });
+
+  return emptyCells[Math.floor(Math.random() * emptyCells.length)];
+}
+
+const btnStart = document.querySelector('.start');
+
+function randomNumber() {
```

Comment:
Consider using more descriptive function names instead of `randomNumber`. For example, `generateRandomNumber`. 🤔